### DISCUSSION TITLE
PING command giving error , "Error: Resource temporarily unavailable" after running process for a few hours.

### DISCUSSION THREAD

**Discussion by shivangi-12** [redis/hiredis#1199] [Q&A]
I am using a daemon process which creates multiple concurrent connections to Redis. I am using [redisConnectWithTimeout] function to connect with Redis with a connect timeout.After running the process for a few hours , the PING command gives this error , "Error: Resource temporarily unavailable" which suggests there is some issue with Redis DB , so we try to reestablish the connection but we get this error everytime. Furthermore , redisFree is giving core dump after we try to free the context just after this error.

reply = redisCommand(*c,"PING");
if(reply == NULL)
{
      printf("command failed");
      printf("Error: %s\n", (*c)->errstr); 
      redisFree (*c); -- This command is giving us core dump
	  
	  // connecting redis back again 
}

**Comment by michael-grunder** [comment]
There's not really enough to go on to help you diagnose the issue.

Can you provide a minimal working example that can replicate the problem you're seeing?

As for the core dump, I would suggest running your program under a debugger or via a tool like `valgrind` to tell you exactly why the crash is occurring.  It could be in hiredis or somewhere else in the program.

> so we try to reestablish the connection

How do you re-establish the connection?
